### PR TITLE
Improve telemetry

### PR DESCRIPTION
**Pull request by 0xOlias**

### CODE REVIEW

File: packages/core/src/common/options.ts [hunk 1]
```diff
@@ -78,9 +78,7 @@ export const buildOptions = ({ cliOptions }: { cliOptions: CliOptions }) => {
 
     telemetryUrl: "https://ponder.sh/api/telemetry",
     telemetryDisabled: Boolean(process.env.PONDER_TELEMETRY_DISABLED),
-    telemetryIsExampleProject: Boolean(
-      process.env.PONDER_TELEMETRY_IS_EXAMPLE_PROJECT,
```
Comment (kyscott18) [inline packages/core/src/common/options.ts:82]: lol never knew we had this

File: packages/core/src/common/telemetry.test.ts [hunk 2]
```diff
@@ -10,30 +13,89 @@ beforeEach(() => {
...
+    telemetry.record({
+      name: "lifecycle:heartbeat_send",
+      properties: { duration_seconds: process.uptime() },
+    });
+  }
+
+  // Mock fetch to take 91ms to complete.
+  fetchSpy.mockImplementation(
+    () => new Promise((resolve) => setTimeout(resolve, 95)),
+  );
```
Comment (kyscott18) [inline packages/core/src/common/telemetry.test.ts:93]: Why 91ms? Also doesn't match the number provider to setTimeout

File: packages/core/src/common/telemetry.ts [hunk 3]
```diff
@@ -1,220 +1,272 @@
...
+      properties: { duration_seconds: process.uptime() },
+    });
+  }, HEARTBEAT_INTERVAL_MS);
+
+  const kill = async () => {
+    clearInterval(heartbeatInterval);
+    await new Promise<void>((resolve) => {
+      if (queue.pending === 0) resolve();
+
+      const timeout = setTimeout(() => {
```
Comment (kyscott18) [inline packages/core/src/common/telemetry.ts:192]: What does this do?

File: packages/core/src/common/telemetry.ts [hunk 4]
```diff
@@ -1,220 +1,272 @@
...
+      const body = JSON.stringify({
+        distinctId: anonymousId,
+        event: task.name,
+        properties,
+      });
 
-  private get anonymousId() {
-    const storedAnonymousId = this.conf.get("anonymousId");
-    if (storedAnonymousId) return storedAnonymousId;
+      await fetch(options.telemetryUrl, {
```
Comment (kyscott18) [inline packages/core/src/common/telemetry.ts:167]: What is the retry behavior if this fails?

File: packages/core/src/common/telemetry.test.ts (hunk 2 above)
Comment (0xOlias) [reply packages/core/src/common/telemetry.test.ts:93]: Oh, these can both be 100 actually now that we're doing >8 and <12 below. I previously had it expecting 10 which appeared flaky.

**Review by kyscott18 — APPROVED** [review]

**Pull request merged**